pkg/tools/tools.go

```diff
@@ -199,6 +170,102 @@ func ImageImportIntoClusterMulti(ctx context.Context, runtime runtimes.Runtime,
+			break
+		}
+
+		runtimeImage, found := findRuntimeImage(requestedImage, runtimeImages)
+		if found {
+			imagesFromRuntime = append(imagesFromRuntime, runtimeImage)
+			log.Debugf("Selected image '%s' (found as '%s') in runtime", requestedImage, runtimeImage)
+			break
+		}
+
+		log.Warnf("Image '%s' is not a file and couldn't be found in the container runtime", requestedImage)
+	}
+	return imagesFromRuntime, imagesFromTar, err
+}
+
+func findRuntimeImage(requestedImage string, runtimeImages []string) (string, bool) {
+	canonicalRequestedImage := canonicalImageName(requestedImage)
+
+	for _, runtimeImage := range runtimeImages {
+		if imageNamesEqual(canonicalRequestedImage, runtimeImage) {
+			return runtimeImage, true
+		}
+	}
+	return "", false
+}
+
+func isFile(image string) bool {
+	file, err := os.Stat(image)
+	if err != nil {
+		return false
+	}
+	return !file.IsDir()
+}
+
+// canonicalImageName turns any docker image name in the form `registry/orgName/imageName:versionName`.
+// If `:versionName` is not present, `:latest` is suffixed.
+// If `orgName` is not present, `library` is added.
+// If `registry` is not present, `docker.io` is prefixed.
+func canonicalImageName(image string) string {
+	if !containsVersionPart(image) {
+		image = fmt.Sprintf("%s:latest", image)
+	}
+
+	slashCount := strings.Count(image, "/")
+	switch slashCount {
+	case 0: // zero slashes -> library image tag (e.g. `postgres`)
+		return fmt.Sprintf("docker.io/library/%s", image)
+	case 1: // one slash -> no registry (e.g. `library/postgres`)
+		return fmt.Sprintf("docker.io/%s", image)
+	default: // two slashes or more -> already contains registry name
+		return image
+	}
+}
+
+func imageNamesEqual(requestedImageName string, runtimeImageName string) bool {
+	if requestedImageName == runtimeImageName {
+		return true
+	}
+
+	return requestedImageName == canonicalImageName(runtimeImageName)
+}
+
+func containsVersionPart(imageTag string) bool {
+	// e.g. imageName has no colon -> false
+	// e.g. repoName/imageName has no colon -> false
+	// e.g. registry/repoName/imageName has no colon -> false
+	// e.g. registry:1234/repoName/imageName has colon, but before the slash -> false
+	// e.g. imageName:versionName has colon -> true
+	// e.g. repoName/imageName:versionName has colon after slash -> true
+	// e.g. registry/repoName/imageName:versionName has colon after slash -> true
+	// e.g. registry:1234/repoName/imageName:versionName has colon after slash -> false
+	if !strings.Contains(imageTag, ":") {
+		return false
+	}
+
+	if !strings.Contains(imageTag, "/") {
+		// happens if someone refers to a library image by just it's imageName (e.g. `postgres` instead of `library/postgres`)
+		return strings.Contains(imageTag, ":")
+	}
+
+	indexOfSlash := strings.Index(imageTag, "/") // can't be -1 because the existence of a '/' is ensured above
+	substringAfterSlash := imageTag[indexOfSlash:]
+	return strings.Contains(substringAfterSlash, ":")
```

Comment:
Wouldn't this bring the same result (last component contains a `:`) (not tested)?
```suggestion
	imageSplit := strings.Split(imageTag, "/")
	retrun strings.Contains(imageSplit[len(imageSplit)-1], ":")
```
Only case in which both would "break" is if someone only provides the registry hostname + port without an image name, but that's simple human error that we shouldn't need to account for.

Reply:
It's true, that this could be shortened like this. The first two if's are just shortcuts. And to me, they're also mental shortcuts. I.e. if the whole string does not have a colon, well then why look any further? From then on, everything else depends on the position of the slash, so if there's not even a slash, again, why look further?

But everything could be done by just looking at the result of `Index(imageTag, "/")` and proceeding from there.

Personnally, I would refrain from using `Split`. As we just need to ensure we're not looking into the hostname, it's only the first slash that matters. So `Index(imageTag, "/")` and then shortening the string to `imageTag[indexOfSlash:]` would do the job perfectly, no need to split all the other parts.